examples/prorata/methods/guest/src/main.rs

```diff
@@ -0,0 +1,22 @@
+#![no_main]
+// If you want to try std support, also update the guest Cargo.toml file
+// #![no_std]  // std support is experimental
+
+use prorata_core::{allocate_for_csv, QueryResult};
+use risc0_zkvm::guest::env;
+use rust_decimal::Decimal;
+
+risc0_zkvm::guest::entry!(main);
+
+pub fn main() {
+    // load the amount, recipients, and target user from the environment
+    let amount: Decimal = env::read();
+    let recipients_csv: Vec<u8> = env::read();
+    let query: String = env::read();
```

Comment:
This would be cleaner to have a struct that is read in, instead of reading in each element separately.

Reply:
Agree that would be cleaner, this is the pattern we demonstrate elsewhere (at least the factors example). Should we update the examples to be single struct in / struct out?
https://github.com/risc0/risc0/blob/3ce67c81ae3fcdc87e6301c2f9371aa53747a3c1/examples/factors/src/lib.rs#L26

Reply:
The factors example is one of (maybe literally) the only examples that uses this multi-read; this is to keep it as barebones as possible to make it as simple of a "Hello World" as possible.

It's possible that's the wrong decision (and I'd be happy to discuss), but it was a deliberate choice.

Reply:
That makes some sense but I think by showing the multi-read pattern in our first example without a clear warning we're effectively endorsing it. The simplest fix would be to put a note in the code and tutorials pointing to an example of the better practice.